builtin/plugins/dposv3/dpos.go

```diff
@@ -292,6 +296,63 @@ func (c *DPOS) Delegate(ctx contract.Context, req *DelegateRequest) error {
 	return c.emitDelegatorDelegatesEvent(ctx, delegation)
 }
 
+//MintVouchers method to the DPOS contract that can be called only by delegators - Mints Loom Coin
+func (c *DPOS) MintVoucher(ctx contract.Context, req *MintVoucherRequest) error {
+	delegations, err := loadDelegationList(ctx)
+	if err != nil {
+		return err
+	}
+	sender := ctx.Message().Sender
+	for _, d := range delegations {
+		if loom.UnmarshalAddressPB(d.Delegator).Compare(sender) == 0 {
+			coin, err := loadCoin(ctx)
+			if err != nil {
+				return err
+			}
+			err = coin.MintToDPOS(&req.Amount.Value)
+			if err != nil {
+				return err
+			}
+			err = coin.TransferFrom(ctx.ContractAddress(), loom.UnmarshalAddressPB(d.Delegator), &req.Amount.Value)
+			if err != nil {
+				transferFromErr := fmt.Sprintf("Failed coin TransferFrom - MintVoucher, %v, %s", ctx.ContractAddress().String(), req.Amount.Value.String())
+				return logDposError(ctx, err, transferFromErr)
+			}
+			break
+		}
+	}
+	return nil
+}
+
+//MintVouchersERC20 method to the DPOS contract that can be called only by delegators - Mints Generic ERC20 Token
+func (c *DPOS) MintVoucherERC20(ctx contract.Context, req *MintVoucherERC20Request) error {
```

Comment:
-> `MintVouchers`

Reply:
Check `req.Amount != nil` and greater than zero before doing anything else.

Reply:
And this new method should return an error if `dpos:v3.6` feature is not enabled.

Reply:
Added nil check for VoucherTokenAddress as well.